torch/csrc/jit/codegen/cuda/fusion.cpp

```diff
@@ -255,7 +261,11 @@ void Fusion::replaceOutput(Val* output, Val* replacement) {
   TORCH_CHECK(find_output != outputs_.end(), "Unable to find output in Fusion");
 
   if (find_output != outputs_.end()) {
-    *find_output = replacement;
+    std::replace_if(
+        outputs_.begin(),
+        outputs_.end(),
+        [&output](Val* v) { return v == output; },
+        replacement);
```

Comment:
Is this changed for when `outputs_` has multiple vals that are equal to `output`?

Reply:
No, I just thought replace_if seemed like better syntax for the operator. I didn't really like the iterator dereference + set. I thought that could have been the issue.

Reply:
If someone sets the same output twice... that would be.... not good.